addons/analytic/models/analytic_plan.py

```diff
@@ -174,6 +174,12 @@ def _compute_children_count(self):
         for plan in self:
             plan.children_count = len(plan.children_ids)
 
+    @api.onchange('parent_id')
+    def _onchange_parent_id(self):
+        project_plan, __ = self._get_all_plans()
+        if self.id.origin == project_plan.id:
```

Comment:
Not sure if it is always safe to assume that the `id` is a `NewId` with an origin.
It isn't if the onchange is called manually at least
```suggestion
        if self._origin.id == project_plan.id:
```

Reply:
Thanks! indeed didn't think of that case. I made the change